File: SwypeEmulatorDemo/SwypeEmulatorDemo/SwypeEmulatorDemo/WordHashContainer.cs
// ***
// *
// * Aurality Studios
// *
// * Swype Algorithm Demo
// *
// * Zackary Misso & Tyler Young
// *
// ***
//
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Content;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Input;

namespace SwypeEmulatorDemo
{
	public class WordHashContainer
	{
		private List<String> words;
		private int code;

		public WordHashContainer(int hash,String first)
		{
			words = new List<String>();
			code=hash;
			words.Add(first);
		}

		// hashcode is the ascii value of the first character in hash
		// concatenated with the hash code of the second character
		public override int GetHashCode()
		{
			return code;
		}

        public List<String> getListOfWords() { words.Sort(); return words; }
		public void Add(String param) { words.Add(param); }
	}
}

File: DictionaryTester/DictionaryTester/src/WordHashContainer.java
//
// Dictionary Controller Test Program
// 
// Aurality Studios
//
// Zack Misso
//
import java.util.ArrayList;
import java.util.Collections;
public class WordHashContainer {
	private ArrayList<String> words;
	private Integer hash;
	
	public WordHashContainer(int code){
		words=new ArrayList<String>();
		hash=code;
	}
	
	public WordHashContainer(int code,String first){
		words=new ArrayList<String>();
		hash=code;
		words.add(first);
	}
	
	public int hashCode(){
		return hash;
	}
	
	public ArrayList<String> getListOfWords(){return words;}
	public void add(String param){words.add(param);}
	
	public String toString(){
		return hash.toString();
	}
}
